holoviews/operation/element.py

```diff
@@ -758,8 +758,12 @@ def _process(self, p, element, ranges={}):
         for d1, d2 in permuted_dims:
             if d1 == d2:
                 if p.diagonal_type is not None:
-                    values = element.dimension_values(d1)
-                    el = p.diagonal_type(values, vdims=[d1])
+                    if p.diagonal_type._1d:
```

Comment:
Any suggestions on naming, ``_add_index`` would come close, but really it's ``_add_integer_index_when_only_1d_array_supplied``.